extensions/functions_comparison.yaml

```diff
@@ -83,4 +83,24 @@ scalar_functions:
       - args:
           - value: fp64
         return: BOOLEAN
+  -
+    name: "is_finite"
+    description: Whether a value is finite (neither infinite nor NaN).
+    impls:
+      - args:
+          - value: fp32
+        return: BOOLEAN
+      - args:
+          - value: fp64
+        return: BOOLEAN
+  -
+    name: "is_infinite"
+    description: Whether a value is infinite.
+    impls:
+      - args:
+          - value: fp32
+        return: BOOLEAN
+      - args:
+          - value: fp64
+        return: BOOLEAN
```

Comment:
I think the `BOOLEAN` and `fp32`/`fp64` need to be nullable since `NULL` would be the return value of a `NULL` input.

After that I think this PR is good to go.

Reply:
Perhaps I am reading the spec wrong but under the [nullability handling section](https://github.com/substrait-io/substrait/blob/3d64e4aa41c9cfb5b5575973d23c07272eb902f9/site/docs/expressions/scalar_functions.md#nullability-handling)

I interpret

> The nullability of the output function is whatever the return type expression states

with `return: BOOLEAN` to mean that no matter whether the input is nullable the output will always be `true` or `false` and never `NULL`.

Am I interpreting that correctly?

Reply:
It looks like nothing else declares `return: T?`, so this is fine according to the current conventions.